pandas/util/_decorators.py

```diff
@@ -268,17 +269,24 @@ def decorator(func: F) -> F:
         def wrapper(*args, **kwargs) -> Callable:
             return func(*args, **kwargs)
 
-        templates = [func.__doc__ if func.__doc__ else ""]
+        # collecting and docstring templates
+        wrapper._doc_args: List[Union[str, Callable]] = []  # type: ignore
```

Comment:
I think the name `_doc_args` is misleading. In the context of this function makes sense, since it's the value of the `args` parameters of the `doc` decorator, but if I see a `_doc_args` attribute of for example `pandas.Series.head`, I don't think it really tells what's in it. Something like `_docstring_components` could be clearer?

Also, when can it contain a callable? Even when we have something like `@doc(pandas.Series.head)` we're extending it with the content templates of `pandas.Series.head`, not with the function.

Last thing, what about using a `docstring_components` variable to avoid all the `type: ignore`, and just set `wrapper._docstring_components = _docstring_components` at the end? Probably cleaner.

Reply:
@datapythonista Thanks for this suggestion, especially for the new variable name. I strongly agree with you here.